### ISSUE TITLE
Imports that are exported are described as unused

### ISSUE THREAD

**Issue by SonofNun15** [palantir/tslint#684]
Given:
```ts
import * as myLib from 'myLib';

export { myLib };
```

tslint gives the following error:
```
app.ts[1, 13]: unused variable: 'myLib'
```

This is a legitimate way to import / export because it allows a component to be exported instead of flattening the hierarchy with a re-export (`export * from 'myLib'`) or exporting the contents of myLib (`export { myLibChild1, myLibChild2 as appChild } from 'myLib';`)

**Comment by adidahiya** [comment]
looks related to #325

**Comment by falsandtru** [comment]
related: https://github.com/Microsoft/TypeScript/issues/5175

also:

```ts
import foo from 'foo';
const bar = {foo};
```

**Comment by blakeembrey** [comment]
Run into this issue too. Same thing as https://github.com/palantir/tslint/issues/818. Has the fix described in https://github.com/palantir/tslint/issues/325#issuecomment-158899660 been tried?

**Comment by SonofNun15** [comment]
To be perfectly honest, we dropped tslint temporarily over this. There was too much churn around TypeScript and ES2015 support. I was planning to incorporate it again once things stabilized and tslint caught up. I'm not aware of the current state / support of tslint.

**Comment by JKillian** [comment]
:+1: on this issue from @zdychacek who was having the same issue in #818 as mentioned by @blakeembrey. 

Agreed this needs to be fixed! If anyone wants to look into it and leave a PR, feel free :)

**Comment by JKillian** [comment]
:+1: from @vasa-c via #897. Copied from his issue:

---

```javascript
import a from "module";

export { a };
```

Error: `file.ts[1, 8]: unused variable: 'a'`.

Next example is OK:

```javascript
import { a } from "module";

export { a };
```

tslint 3.2.1
tsc 1.7.5

**Comment by JKillian** [comment]
It looks like there'll be a fix in TypeScript for this shortly! https://github.com/Microsoft/TypeScript/pull/6376

**Comment by adidahiya** [comment]
@JKillian let's add tests for this on the `next` branch now that the compiler bug is fixed.

**Comment by JKillian** [comment]
This should be fixed now if you use `typescript@1.8.0-dev.20160116` or newer alongside of TSLint.
